gstreamer-pbutils/src/auto/functions.rs

```diff
@@ -0,0 +1,373 @@
+    unsafe {
+        from_glib_full(ffi::gst_missing_decoder_installer_detail_new(decode_caps.to_glib_none().0))
+    }
+}
+
+pub fn missing_decoder_message_new<P: IsA<gst::Element>>(element: &P, decode_caps: &gst::Caps) -> Option<gst::Message> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_decoder_message_new(element.to_glib_none().0, decode_caps.to_glib_none().0))
+    }
+}
+
+pub fn missing_element_installer_detail_new(factory_name: &str) -> Option<String> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_element_installer_detail_new(factory_name.to_glib_none().0))
+    }
+}
+
+pub fn missing_element_message_new<P: IsA<gst::Element>>(element: &P, factory_name: &str) -> Option<gst::Message> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_element_message_new(element.to_glib_none().0, factory_name.to_glib_none().0))
+    }
+}
+
+pub fn missing_encoder_installer_detail_new(encode_caps: &gst::Caps) -> Option<String> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_encoder_installer_detail_new(encode_caps.to_glib_none().0))
+    }
+}
+
+pub fn missing_encoder_message_new<P: IsA<gst::Element>>(element: &P, encode_caps: &gst::Caps) -> Option<gst::Message> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_encoder_message_new(element.to_glib_none().0, encode_caps.to_glib_none().0))
+    }
+}
+
+pub fn missing_plugin_message_get_description(msg: &gst::Message) -> Option<String> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_plugin_message_get_description(msg.to_glib_none().0))
+    }
+}
+
+pub fn missing_plugin_message_get_installer_detail(msg: &gst::Message) -> Option<String> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_plugin_message_get_installer_detail(msg.to_glib_none().0))
+    }
+}
+
+pub fn missing_uri_sink_installer_detail_new(protocol: &str) -> Option<String> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_uri_sink_installer_detail_new(protocol.to_glib_none().0))
+    }
+}
+
+pub fn missing_uri_sink_message_new<P: IsA<gst::Element>>(element: &P, protocol: &str) -> Option<gst::Message> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_uri_sink_message_new(element.to_glib_none().0, protocol.to_glib_none().0))
+    }
+}
+
+pub fn missing_uri_source_installer_detail_new(protocol: &str) -> Option<String> {
+    assert_initialized_main_thread!();
+    unsafe {
+        from_glib_full(ffi::gst_missing_uri_source_installer_detail_new(protocol.to_glib_none().0))
+    }
+}
+
+pub fn missing_uri_source_message_new<P: IsA<gst::Element>>(element: &P, protocol: &str) -> Option<gst::Message> {
```

Comment:
All the missing plugin stuff should be hidden: messages should be done as in core. Or as the video events. Not just raw functions

Generally, hide all the functions that you don't want to expose :)